services/core-api/app/api/services/ams_api_service.py

```diff
@@ -95,83 +100,119 @@ def create_new_ams_authorization(cls,
-                    'facilitydescription': facility_desc,
-                    'facilitylocationlatitude': str(facility_latitude),
-                    'facilitylocationlongitude': str(facility_longitude),
-                    'sourceofdata': facility_coords_source,
-                    'sourceofdatadescription': facility_coords_source_desc,
-                    'legallanddescription': legal_land_desc,
-                    'facilityaddress': {
-                        'addresstype': 'Civic',
-                        'suitenumber': facility_operator.get('address').get('suite_no', ''),
-                        'streetnumber': facility_operator.get('address').get('suite_no', ''),
-                        'street': facility_operator.get('address').get('address_line_1', ''),
-                        'line2': facility_operator.get('address').get('address_line_2', ''),
-                        'municipality': facility_operator.get('address').get('city', ''),
-                        'province': 'British Columbia',
-                        'country': 'Canada',
-                        'postalcode': facility_operator.get('address').get('post_code')
-                    },
-                    'facilityopname': facility_operator.get('name', ''),
-                    'facilityopphonenumber': cls.__format_phone_number(facility_operator.get('phone_no', '')),
-                    'facilityopphonenumberext': facility_operator.get('phone_ext', ''),
-                    'facilityopemail': facility_operator.get('email', ''),
-                    'landownername': legal_land_owner_name,
-                    'landownerphonenumber': cls.__format_phone_number(legal_land_owner_contact_number),
-                    'landowneremail': legal_land_owner_email_address,
-                    'istheapplicantthelandowner': cls.__boolean_to_yes_no(is_legal_land_owner),
-                    'landfedorprov': cls.__boolean_to_yes_no(is_crown_land_federal_or_provincial),
-                    'landownerawareofapplication': cls.__boolean_to_yes_no(is_landowner_aware_of_discharge_application),
-                    'landownerreceivedcopy': cls.__boolean_to_yes_no(has_landowner_received_copy_of_application)
-                }
-                payload = json.dumps(ams_authorization_data)
-                response = requests.post(Config.AMS_URL, data=payload, headers=headers)
-                ams_result = response.json()
-                ams_result['project_summary_authorization_guid'] = authorization.get('project_summary_authorization_guid')
-                ams_result['project_summary_guid'] = authorization.get('project_summary_guid')
-                ams_result['project_summary_authorization_type'] = authorization.get(
-                    'project_summary_authorization_type')
-                ams_results.append(ams_result)
+                existing_ams_status_code = authorization.get('ams_status_code')
+                if existing_ams_status_code != '200':
```

Comment:
This is to prevent the user from submitting to AMS again if the previous submission was successful.